web/src/app/chat/ChatPage.tsx

```diff
@@ -244,7 +250,13 @@ export function ChatPage({
       const newMessageMap = processRawChatHistory(chatSession.messages);
       const newMessageHistory = buildLatestMessageChain(newMessageMap);
       // if the last message is an error, don't overwrite it
-      if (messageHistory[messageHistory.length - 1]?.type !== "error") {
+
+      if (
+        !(
```

Comment:
nit: prefer to remove the `!` and just flip each condition internally. This feels pretty hard to understand currently